### ISSUE TITLE
sqlc does not check if the field after `update set` statement exists or not

### ISSUE THREAD

**Issue by immysec** [sqlc-dev/sqlc#1128]
Input:

```sql
-- Example queries for sqlc
CREATE TABLE Servers (
  code varchar    PRIMARY KEY,
  name text      NOT NULL
);

-- name: UpsertServer :exec
insert into Servers(code, name) values ($1, $2) on conflict do update set name_typo = 1111;
```

Output:
```go
// Code generated by sqlc. DO NOT EDIT.
// source: query.sql

package db

import (
	"context"
)

const upsertServer = `-- name: UpsertServer :exec
insert into Servers(code, name) values ($1, $2) on conflict do update set name_typo = 1111
`

type UpsertServerParams struct {
	Code string
	Name string
}

func (q *Queries) UpsertServer(ctx context.Context, arg UpsertServerParams) error {
	_, err := q.db.ExecContext(ctx, upsertServer, arg.Code, arg.Name)
	return err
}
```

I often rename field name in table definition but always forget to rename the fields after `update set`. I thought sqlc will error me the mismatch...

Here is the [Playground](https://play.sqlc.dev/p/fc8eac55505c2798ce9a0e395e71b3366b455a3956d41b0a4af067d37b7851fa).

**Comment by brentd** [comment]
@kyleconroy I'd like to implement support for this - is this something that should be easy to implement or is there missing information in the ast?

**Comment by kyleconroy** [comment]
This is fixed in [v1.23.0](https://docs.sqlc.dev/en/latest/reference/changelog.html#v1-23-0) by enabling the [database-backed query analyzer](https://docs.sqlc.dev/en/latest/howto/generate.html#enhanced-analysis-with-managed-databases). We added a [test case](https://github.com/sqlc-dev/sqlc/tree/main/internal/endtoend/testdata/update_set_on_conflict/postgresql/pgx) for this issue so it won’t break in the future.

You can play around with the [working example on the playground](https://play.sqlc.dev/p/c65659c988f3346e9c97c9145fd4d75d3792463fa66741800c0b6cc94111ddc8)
